exercises/crypto-square/makefile

```diff
@@ -0,0 +1,25 @@
+CFLAGS  = -std=c99
+CFLAGS += -g
+CFLAGS += -Wall
+CFLAGS += -Wextra
+CFLAGS += -pedantic
+CFLAGS += -Werror
+
+VFLAGS  = --quiet
+VFLAGS += --tool=memcheck
+VFLAGS += --leak-check=full
+VFLAGS += --error-exitcode=1
+
+test: tests.out
+	@./tests.out
+
+memcheck: tests.out
+	@valgrind $(VFLAGS) ./tests.out
+	@echo "Memory check passed"
+
+clean:
+	rm -rf *.o *.out *.out.dSYM
+
+tests.out: test/test_crypto_square.c src/crypto_square.c src/crypto_square.h
+	@echo Compiling $@
+	@cc $(CFLAGS) src/crypto_square.c test/vendor/unity.c test/test_crypto_square.c -o tests.out -lm
```

Comment:
Ideally the `-lm` should have used a `LDFLAGS` variable here. As it is merged is it probably OK as-is for now but maybe bear in mind in the future :smiley: